File: 0055-jump-game/0055-jump-game.c
bool canJump(int* nums, int numsSize){
    bool tab[numsSize];
    tab[0]=true;
    for(int i=1;i<numsSize;i++){
        tab[i]=false;
        for(int j=i-1;j>=0;j--){
            tab[i]=(tab[j] && nums[j]>=i-j )&& nums[j]>0;
            if(tab[i]) break;
        }
    }
    return tab[numsSize-1];
}

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        maxIndex = 0
        
        for i in range(len(nums)-1):
            if maxIndex<i:
                return False
            if i+nums[i]>maxIndex:
                maxIndex = i +nums[i]
            if maxIndex>=len(nums)-1:
                return True
        return maxIndex >= len(nums)-1